benchmark/benchmark_shallow_water_model.jl

```diff
@@ -0,0 +1,43 @@
+using BenchmarkTools
+using CUDA
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+using Benchmarks
+
+# Benchmark function
+
+function benchmark_shallow_water_model(Arch, FT, N)
+    grid = RegularRectilinearGrid(FT, size=(N, N, 1), extent=(1, 1, 1))
+    model = ShallowWaterModel(architecture=Arch(), grid=grid, gravitational_acceleration=1.0)
+    set!(model, h=1)
+
+    time_step!(model, 1) # warmup
+
+    trial = @benchmark begin
+        @sync_gpu time_step!($model, 1)
+    end samples=10
+    
+    return trial
+end
+
+# Benchmark parameters
+
+Architectures = has_cuda() ? [CPU, GPU] : [CPU]
+#Architectures = [CPU, GPU, MultiCPU, MultiGPU]
```

Comment:
Sorry I forgot to cut this.  We probably should remove this comment

Reply:
Done.